cpp/source/deal.II/physics.h

```diff
@@ -0,0 +1,92 @@
+#ifndef CAP_PHYSICS_H
+#define CAP_PHYSICS_H
+
+#include <cap/boundary_values.h>
+#include <cap/mp_values.h>
+#include <boost/property_tree/ptree.hpp>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/sparsity_pattern.h>
+#include <deal.II/lac/vector.h>
+
+namespace cap
+{
+/**
+ * This class encapsulate all the parameters needed to build a Physics object.
+ */
+template <int dim>
+class PhysicsParameters
+{
+public:
+  PhysicsParameters(std::shared_ptr<boost::property_tree::ptree const> d)
+      : database(d)
+  {
+  }
+
+  virtual ~PhysicsParameters() = default;
+
+  std::shared_ptr<dealii::DoFHandler<dim> const> dof_handler;
+  std::shared_ptr<MPValues<dim> const> mp_values;
+  std::shared_ptr<BoundaryValues<dim> const> boundary_values;
+  std::shared_ptr<boost::property_tree::ptree const> database;
+};
+
+/**
+ * This is the base class for the Physics. The derived Physics need to build
+ * the matrix of the system and the right-hand side. They are also responsible
+ * of the boundary conditions.
+ */
+template <int dim>
+class Physics
+{
+public:
+  Physics(std::shared_ptr<PhysicsParameters<dim> const> parameters);
```

Comment:
We need to keep this one because ```ElectrochemicalPhysicsParameters``` derives from it and it allows us to use polymorphism.